pkg/gitops.go

```diff
@@ -283,23 +296,24 @@ func GenerateOverlaysAndPush(outputPath string, clone bool, remote string, optio
 // 6. The branch to push to
 // 7. The path within the repository to generate the resources in
 // 8. Optionally push the changes to the repository
-func RemoveAndPush(outputPath string, remote string, componentName string, e Executor, appFs afero.Afero, branch string, context string, doPush bool) error {
+func RemoveComponent(outputPath string, remote string, componentName string, e Executor, appFs afero.Afero, branch string, context string, doClone, doPush bool) error {
 
 	invalidRemoteErr := util.ValidateRemote(remote)
 	if invalidRemoteErr != nil {
 		return invalidRemoteErr
 	}
 
-	if out, err := e.Execute(outputPath, "git", "clone", remote, componentName); err != nil {
-		return fmt.Errorf("failed to clone git repository in %q %q: %s", outputPath, string(out), err)
-	}
-
 	repoPath := filepath.Join(outputPath, componentName)
 
-	// Checkout the specified branch
-	if _, err := e.Execute(repoPath, "git", "switch", branch); err != nil {
-		if out, err := e.Execute(repoPath, "git", "checkout", "-b", branch); err != nil {
-			return fmt.Errorf("failed to checkout branch %q in %q %q: %s", branch, repoPath, string(out), err)
+	if doClone {
+		if out, err := e.Execute(outputPath, "git", "clone", remote, componentName); err != nil {
+			return fmt.Errorf("failed to clone git repository in %q %q: %s", outputPath, string(out), err)
+		}
+		// Checkout the specified branch
+		if _, err := e.Execute(repoPath, "git", "switch", branch); err != nil {
```

Comment:
should this be outside the if condition, what if the repo is already cloned and on a different branch

Reply:
Good point. The intent of this is really to not do both the clone and switch.  Just wanted to remove what is present in the filesystem at that time.  So the boolean flag could be renamed to doCloneAndSwitch from doClone.  This is similar to the call to `CommitAndPush` where both the commit and push git commands are run, but the flag is named `doPush`.

Alternatively, we could have yet another bool: doClone and doSwitch to go along with the existing doPush.

Reply:
@maysunfaisal , now that we have the refactoring and simplification of types done, and the original APIs are moved over to this repo, I wanted to improve them.  eg. one way is to try to improve the names, starting with this function, so changing the name from RemoveAndPush to simply RemoveComponent.  We will eventually have RemoveEnvironment (or overlay) too.

Reply:
@maysunfaisal , please review this.  I have made the API more cleaner and eliminated the need for the doPush and doClone boolean flags.